docs/add_a_validation.md

```diff
@@ -0,0 +1,295 @@
+
+``` php
+// eZ/Publish/FieldType/Tweet/Type.php
+
+public function validateValidatorConfiguration($validatorConfiguration)
+{
+    $validationErrors = [];
+
+    foreach ($validatorConfiguration as $validatorIdentifier => $constraints) {
+        // Report unknown validators
+        if ($validatorIdentifier !== 'TweetValueValidator') {
+            $validationErrors[] = new ValidationError("Validator '$validatorIdentifier' is unknown");
+            continue;
+        }
+
+        // Validate arguments from TweetValueValidator
+        foreach ($constraints as $name => $value) {
+            switch ($name) {
+                case 'authorList':
+                    if (!is_array($value)) {
+                        $validationErrors[] = new ValidationError('Invalid authorList argument');
+                    }
+                    foreach ($value as $authorName) {
+                        if (!preg_match('/^[a-z0-9_]{1,15}$/i', $authorName)) {
+                            $validationErrors[] = new ValidationError('Invalid twitter username');
+                        }
+                    }
+                    break;
+                default:
+                    $validationErrors[] = new ValidationError("Validator parameter '$name' is unknown");
+            }
+        }
+    }
+
+    return $validationErrors;
+}
+```
+
+`validate()` is the method that runs the actual validation on data:
+
+``` php
+// eZ/Publish/FieldType/Tweet/Type.php
+
+public function validate(FieldDefinition $fieldDefinition, SPIValue $fieldValue)
+{
+    $errors = [];
+
+    if ($this->isEmptyValue($fieldValue)) {
+        return $errors;
+    }
+
+    // Tweet Url validation
+    if (!preg_match('#^https?://twitter.com/([^/]+)/status/[0-9]+$#', $fieldValue->url, $m)) {
+        $errors[] = new ValidationError(
+            'Invalid twitter status url %url%',
+            null,
+            ['%url%' => $fieldValue->url]
+        );
+
+        return $errors;
+    }
+
+    $author = $m[1];
+    $validatorConfiguration = $fieldDefinition->getValidatorConfiguration();
+    if (!$this->isAuthorApproved($author, $validatorConfiguration)) {
+        $errors[] = new ValidationError(
+            'Twitter user %user% is not in the approved author list',
+            null,
+            ['%user%' => $m[1]]
+        );
+    }
+
+    return $errors;
+}
+
+private function isAuthorApproved($author, $validatorConfiguration)
+{
+    return !isset($validatorConfiguration['TweetValueValidator'])
+        || empty($validatorConfiguration['TweetValueValidator']['authorList'])
+        || in_array($author, $validatorConfiguration['TweetValueValidator']['authorList']);
+}
+```
+
+In addition to validating the URL with a regular expression, which we implemented before, if your Field Type instance’s configuration contains a `TweetValueValidator` key, you will check that the username in the status URL matches one of the valid authors.
```

Comment:
A bit longing, I'd rephrase it like this:
Earlier we validated the URL with a regular expression. Now, if your the configuration of your Field Type's instance contains a `TweetValueValidator` key, you will check that the username in the status URL matches one of the valid authors.